myapp/app/controllers/tasks_controller.rb

```diff
@@ -1,13 +1,12 @@
 class TasksController < ApplicationController
   before_action :set_task, only: [:show, :edit, :update, :destroy]
   
+  PER_PAGE = 10
+  
   def index
-    if %w[created_at deadline].include?(params[:sort_by])
-      direction = %w[asc desc].include?(params[:direction]) ? params[:direction] : 'asc'
-      @tasks = Task.order(params[:sort_by] => direction)
-    else
-      @tasks = Task.order(:id)
-    end
+    @q = Task.ransack(params[:q])
+    @q.sorts = 'created_at asc' if @q.sorts.empty?
+    @tasks = @q.result(distinct: true).page(params[:page]).per(PER_PAGE)
```

Comment:
Since the search results don't seem to have any duplicates, the `distinct` might be redundant. (Let me know if I've missed something.)
Removing unnecessary code like this can improve code clarity.